platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/ProfileInstallCommand.java

```diff
@@ -0,0 +1,257 @@
+    }
+
+    private void stopBundles(BundleService bundleService, List<String> stopBundleNames)
+            throws BundleException, IllegalArgumentException{
+        if (stopBundleNames != null) {
+            printSectionHeading("Stopping Bundles");
+            stopBundleNames.stream().distinct().forEach(bundle -> {
+                printItemStatusPending("Stopping: ", bundle);
+                executeAsSystem(() -> {
+                    try {
+                        bundleService.getBundle(bundle).stop();
+                    } catch (BundleException e) {
+                        printItemStatusFailure("Stop Failed: ", bundle);
+                        throw e;
+                    } catch (IllegalArgumentException ie) {
+                        printItemStatusFailure("Stop Failed: ", bundle);
+                        printError(String.format("Bundle: %s does not exist", bundle));
+                        throw ie;
+                    }
+                    return true;
+                });
+                printItemStatusSuccess("Stopped: ", bundle);
+            });
+        }
+    }
+
+    /**
+     * Get the {@link Feature} that makes up a profile
+     * @param applicationService {@link ApplicationService}
+     * @param profileName Name of profile to get
+     * @return {@link Optional} if the profile doesn't exist the {@link Optional} will be empty
+     */
+    private Optional<Feature> getProfile(ApplicationService applicationService, String profileName) {
+        return applicationService.getInstallationProfiles().stream()
+                .filter(i -> i.getName().replaceFirst(PROFILE_PREFIX, "").equals(profileName)).findFirst();
+    }
+
+    /**
+     * Gets the content of an advanced profile
+     * @param profileName Name of profile to get
+     * @return {@link Optional}, if the profile doesn't exist the {@link Optional} will be empty
+     */
+    private Optional<Map<String, List<String>>> getProfile(String profileName) throws IOException {
+        File profileFile = Paths.get(profilePath.toString(), profileName + PROFILE_EXTENSION).toFile();
+        Map<String, List<String>> profileMap = null;
+
+        try {
+            profileMap = (Map<String, List<String>>) fromJson(FileUtils.readFileToString(profileFile, StandardCharsets.UTF_8));
+        } catch (FileNotFoundException e) {
+            LOGGER.debug("The file associated with profile: {} was not found under {}",
+                    profileName, profilePath);
+        } catch (IOException e) {
+            LOGGER.debug("An IOException occurred: ", e);
+        }
+
+        return Optional.ofNullable(profileMap);
+    }
+
+    private <T> T executeAsSystem(Callable<T> func) {
+        Subject systemSubject = security.getSystemSubject();
+        LOGGER.debug("System Subject retrieved: " + SubjectUtils.getName(systemSubject));
+        if (systemSubject == null) {
+            printError(SECURITY_ERROR);
+            throw new RuntimeException(SECURITY_ERROR);
```

Comment:
❗️ Even though other commands do this, we should use something else than `RuntimeException` here, e.g., `IllegalStateException` or such.